docs/DeveloperGuide.md

```diff
@@ -1,34 +1,845 @@
+
+-   `Storage`: Reads and stores data from a file stored on the user’s computer
+
+The following sections will explain each component in greater detail.
+
+### 2.2. UI Component
+1.  The `Ui` component reads user input which represents a command.
+
+2.  The `Ui` component shows the result of the command to the user.
+
+### 2.3. Logic Component
+
+The Logic component comprises the `Parser`, `Command`, and `CommandResult` classes:  
+
+1.  `Parser` will interpret the user command and return the corresponding `Command` object.  
+
+2.  `Command#execute()` is called to run the command, returning a `CommandResult` object.  
+
+3.  The `CommandResult` object will contain the output to be shown to the user.
+
+### 2.4. Model Component
+The Model component contains the `Task` and `TaskList` classes, which store the user’s schedule.
+
+![TaskList and Tasks](images/TaskList Task class diagram.PNG)
+
+### 2.5. Storage Component
+![Storage Class Diagram](images/storage.PNG)
+
+1.  A `Storage` object is created by the `Atas` class to handle the loading and saving of `Task` data.
+
+2.  The `load()` method is used to read saved data from a local file into the current session of **ATAS**.
+
+3.  The `save()` method writes the current state of **ATAS** into the local save file using the `Task#encodeTask()` method.
+
+### 2.6. Atas Component
+The `Atas` component integrates all the aforementioned components to run the overall application logic.  
+The sequence diagram below shows how various components, broken down into the various classes, interact when the user enters a `help` command  
+
+![Component interactions for help command](images/atas help command sequence diagram v3.PNG)
+
+1.  The `Ui` class is used to read user input.  
+
+2.  The `Parser` class is used to parse the user input string, returning a `Command` object.  
+
+3.  The `Command#execute()` method is run, returning a `CommandResult` object.  
+
+4.  The `Ui` class is used to show the `CommandResult` message to the user.  
+
+5.  The `Storage` object is used to save the new state of the application.
+
+## 3. Setting Up
+- For **Windows** Users:
+    1. Download the latest release of ATAS [here](https://github.com/AY1920S2-CS2113T-M16-1/tp/releases).
+    2. Open a ```cmd```(Command Prompt) window.
+    3. Navigate to the directory containing downloaded jar file.
+    4. Run the command ```java -jar atas.jar```. You will be greeted with the welcome screen of ATAS in a few seconds.
+- For **Mac** Users:
+    1. Download the latest release of ATAS [here](https://github.com/AY1920S2-CS2113T-M16-1/tp/releases).
+    2. Open up ```Terminal```
+    3. Navigate to the directory containing downloaded jar file.
+    4. Run the command ```java -jar atas.jar```. You will be greeted with the welcome screen of ATAS in a few seconds.
+
+## 4. Implementation
+This section will detail how some noteworthy features are implemented.
+### 4.1. Delete Task Feature
+Current Implementation:  
```

Comment:
Should there be a header for current implementation - ie. 4.1.1, as it divided into a subsection under search task feature, as section 4.2.1